projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -64,147 +65,114 @@ public static function build_view_data() {
 	 */
-	public static function get_footer_to_add( $data ) {
-		if ( self::is_amp_request() ) {
-			/**
-			 * Filter the parameters added to the AMP pixel tracking code.
-			 *
-			 * @module stats
-			 *
-			 * @since-jetpack 10.9
-			 *
-			 * @param array $data Array of options about the site and page you're on.
-			 */
-			$data = (array) apply_filters( 'jetpack_stats_footer_amp_data', $data );
-			return self::get_amp_footer( $data );
-		} else {
-			/**
-			 * Filter the parameters added to the JavaScript stats tracking code.
-			 *
-			 * @module stats
-			 *
-			 * @since-jetpack 10.9
-			 *
-			 * @param array $data Array of options about the site and page you're on.
-			 */
-			$data = (array) apply_filters( 'jetpack_stats_footer_js_data', $data );
-			return self::get_footer( $data );
-		}
+	private static function get_amp_footer( $data ) {
+		$data['host'] = isset( $_SERVER['HTTP_HOST'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_HOST'] ) ) : ''; // input var ok.
+		$data['rand'] = 'RANDOM'; // AMP placeholder.
+		$data['ref']  = 'DOCUMENT_REFERRER'; // AMP placeholder.
+		$data         = array_map( 'rawurlencode', $data );
+		$pixel_url    = add_query_arg( $data, 'https://pixel.wp.com/g.gif' );
+		return '<amp-pixel src="' . esc_url( $pixel_url ) . '"></amp-pixel>';
 	}
 
 	/**
-	 * Get the list of attributes to add to the Stats script tag.
+	 * Build an AMP pixel.
 	 *
-	 * @since $$next-version$$
-	 *
-	 * @return string string of attributes to add to the script tag.
+	 * @return void
 	 */
-	public static function get_script_attributes() {
-		/**
-		 * Allow adding extra attributes to the stats footer script tag.
-		 *
-		 * @module stats
-		 *
-		 * @since $$next-version$$
-		 *
-		 * @param array $script_attributes Array of attributes to add to the script tag. Default to 'defer' only.
-		 */
-		$script_attributes = (array) apply_filters(
-			'jetpack_stats_footer_script_attributes',
-			array( 'defer' => true )
-		);
-
-		/*
-		 * Create an escaped string of attributes to add to the script tag,
-		 * like 'defer', 'async',
-		 * or anything passed to the filter above.
-		 *
-		 * For boolean attributes, we either only display the key (e.g. "defer")
-		 * or not display anything (e.g. we do not want to display "async=false").
-		 */
-		$atts_to_add = array();
-		foreach ( $script_attributes as $key => $value ) {
-			if ( is_bool( $value ) ) {
-				if ( $value ) {
-					$atts_to_add[] = esc_attr( $key );
-				}
-			} else {
-				$atts_to_add[] = esc_attr( $key ) . "='" . esc_attr( $value ) . "'";
-			}
+	public static function add_amp_pixel() {
+		$data = self::build_view_data();
+		if ( self::is_amp_request() ) {
+			return;
 		}
-		$script_attributes = implode( ' ', $atts_to_add );
 
-		return $script_attributes;
+		$pixel = self::get_amp_footer( $data );
+		echo $pixel; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 	}
 
 	/**
-	 * Gets the stats footer
+	 * Stats Footer.
 	 *
```

Comment:
`@deprecated` too? And same for the rest being deprecated here?

Reply:
👍 d1868ad5e512d2327f145dd58b4324a0e749f621